test/e2e/v3/plugin_cluster_test.go

```diff
@@ -122,64 +90,18 @@ var _ = Describe("kubebuilder", func() {
-					Skip(fmt.Sprintf("cluster version %s does not support v1 CRDs or webhooks",
-						srvVer.GitVersion))
-				}
-
 				GenerateV3(kbc, "v1")
 				Run(kbc)
 			})
-			It("should generate a runnable project with v1beta1 CRDs and Webhooks", func() {
-				// Skip if cluster version < 1.15, when `.spec.preserveUnknownFields` was not a v1beta1 CRD field.
-				// Skip if cluster version >= 1.22 because pre v1 CRDs and webhooks no longer exist.
-				if srvVer := kbc.K8sVersion.ServerVersion; srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() < 15 ||
-					srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() >= 22 {
-					Skip(fmt.Sprintf("cluster version %s does not support project defaults ",
-						srvVer.GitVersion))
-				}
-
-				GenerateV3(kbc, "v1beta1")
-				Run(kbc)
-			})
 
 			It("should generate a runnable project go/v3 with v1 CRDs and Webhooks with restricted pods", func() {
-				// Skip if cluster version < 1.16, when v1 CRDs and webhooks did not exist.
-				// Skip if cluster version < 1.19, because securityContext.seccompProfile only works from 1.19
-				// Otherwise, unknown field "seccompProfile" in io.k8s.api.core.v1.PodSecurityContext will be faced
-				if srvVer := kbc.K8sVersion.ServerVersion; srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() < 19 {
-					Skip(fmt.Sprintf("cluster version %s does not support v1 CRDs or webhooks"+
-						"and securityContext.seccompProfile", srvVer.GitVersion))
-				}
-
 				kbc.IsRestricted = true
 				GenerateV3(kbc, "v1")
 				Run(kbc)
 			})
-			It("should generate a runnable project with v1beta1 CRDs and Webhooks with restricted pods", func() {
-				// Skip if cluster version < 1.15, when `.spec.preserveUnknownFields` was not a v1beta1 CRD field.
-				// Skip if cluster version < 1.19, because securityContext.seccompProfile only works from 1.19
-				// Otherwise, unknown field "seccompProfile" in io.k8s.api.core.v1.PodSecurityContext will be faced
-				// Skip if cluster version >= 1.22 because pre v1 CRDs and webhooks no longer exist.
-				if srvVer := kbc.K8sVersion.ServerVersion; srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() < 19 ||
-					srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() >= 22 {
-					Skip(fmt.Sprintf("cluster version %s does not support project defaults "+
-						"and securityContext.seccompProfile", srvVer.GitVersion))
-				}
 
-				kbc.IsRestricted = true
-				GenerateV3(kbc, "v1beta1")
-				Run(kbc)
-			})
 			It("should generate a runnable project go/v3 with v1 CRDs and Webhooks"+
 				"with --component-config flag enabled", func() {
-				// Skip if cluster version < 1.16, when v1 CRDs and webhooks did not exist.
-				if srvVer := kbc.K8sVersion.ServerVersion; srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() < 16 {
-					Skip(fmt.Sprintf("cluster version %s does not support v1 CRDs or webhooks",
-						srvVer.GitVersion))
-				}
-
 				GenerateV3ComponentConfig(kbc, "v1")
```

Comment:
@Eileen-Yu,

That is great. Just a nit. 
See that we no longer need to pass v1 or v1beta1
So, we can clean up the calls as well removing this option as the related code.

Example: https://github.com/kubernetes-sigs/kubebuilder/blob/ed5d217ba5fae4fdf7ef63a20fa6de34ef359384/test/e2e/v3/generate_test.go#L133